cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -144,3 +145,62 @@ void sc_check_rootfs_dir(sc_invocation *inv) {
 
     die("cannot locate base snap %s", inv->base_snap_name);
 }
+
+static char* read_homedirs_from_system_params(void)
+{
+    FILE *f SC_CLEANUP(sc_cleanup_file) = NULL;
+    f = fopen("/var/lib/snapd/system-params", "r");
+    if (f == NULL) {
+        return NULL;
+    }
+
+    char *line SC_CLEANUP(sc_cleanup_string) = NULL;
+    size_t line_size = 0;
+    while (getline(&line, &line_size, f) != -1) {
+        if (sc_startswith(line, "homedirs=")) {
+            return sc_strdup(line + (sizeof("homedirs=") - 1));
+        }
+    }
+    return NULL;
+}
+
+void sc_invocation_check_homedirs(sc_invocation *inv)
```

Comment:
since this modifies `inv` by setting up `homedirs` should it instead be called something like `sc_invocation_populate_homedirs`?

Reply:
Yea that makes much more sense,  I renamed it to `sc_invocation_init_homedirs` to be in line with other namings